File: Python/numIdenticalPairs.py


from typing import List


class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        count = 0
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i] == nums[j]:
                    count += 1
        return count

def main():
    solution = Solution()
    #user_input = input()
    #nums = [int(x) for x in user_input.split(",")]
    nums = [1,2,3,1,1,3]
    result = solution.numIdenticalPairs(nums)
    print(result)

if __name__ == "__main__":
    main()

File: Newbie_basics/C++/numIdenticalPairs.cpp
#include <iostream>
using namespace std;
#include <vector>
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int count = 0;
        for (int i = 0; i < nums.size(); i++) {
            for (int j = i + 1; j < nums.size(); j++) {
                if (nums[i] == nums[j]) {
                    count++;
                }
            }
        }
        return count;
    }
};
int main() {
    Solution solution;
    vector<int> nums = {1, 2, 3, 1, 1, 3};
    cout << solution.numIdenticalPairs(nums) << endl;
    return 0;
}